vmdb/app/models/ems_microsoft.rb

```diff
@@ -96,7 +96,8 @@ def execute_power_operation(cmdlet, vm_uid_ems, *parameters)
     return unless vm_uid_ems.guid?
 
     params  = parameters.join(" ")
-    command = "powershell #{cmdlet}-SCVirtualMachine -VM (Get-SCVirtualMachine -ID #{vm_uid_ems}) #{params}"
+    command = "powershell Import-Module VirtualMachineManager; Get-VMMServer #{ipaddress}; "
```

Comment:
@jerryk55 Let me think about this... I really don't want us doing IPv6 checking in our code. The URI library can do this for us and we should use it.

I'm thinking we need a utility method to do something like this for us:

```ruby
irb(main):001:0> def uri_hostname(host)
irb(main):002:0>   require 'uri'
irb(main):003:1>   u = URI("http://foo")
irb(main):004:1>   u.hostname = host  # ipv6 safe method to set an IPv6 literal
irb(main):005:1>   u.host
irb(main):006:1> end
=> :uri_hostname
irb(main):007:0> uri_hostname("localhost")
=> "localhost"
irb(main):008:0> uri_hostname("::1")
=> "[::1]"
irb(main):009:0> uri_hostname("127.0.0.1")
=> "127.0.0.1"
```

@Fryguy @tenderlove what are your thoughts?

I have other "hacky" workarounds for IPv6 URI building that can also live in a single place instead of having these :hankey: things littered all over our codebase.